qutebrowser/config/config.py

```diff
@@ -637,14 +638,22 @@ class StyleSheetObserver(QObject):
 
     """Set the stylesheet on the given object and update it on changes.
 
+    Note that if update is turned on and the stylesheet accesses any config
+    option with a syntax different from 'conf.abc.def' (with surrounding
+    spaces) then an explicit option 'options' should be passed to be used as
+    self._options.
+
     Attributes:
         _obj: The object to observe.
         _stylesheet: The stylesheet template to use.
+        _options: The config options that the stylesheet uses. When it's not
+                  necessary to listen for config changes, this attribute may be
+                  missing.
     """
 
     def __init__(self, obj: QObject,
-                 stylesheet: typing.Optional[str],
-                 update: bool) -> None:
+                 stylesheet: Optional[str], update: bool,
+                 options: Optional[FrozenSet[str]] = None) -> None:
```

Comment:
It might be a little cleaner to avoid Optional here and just default to a empty frozenset instead. That way if the update hook runs improperly it won't crash on `for a in None`

Reply:
That's different, None means "parse from the template", empty means "this doesn't depend	on anything although the template may suggest otherwise".